src/gtk/window.cpp

```diff
@@ -5077,23 +5083,23 @@ void wxWindowGTK::GTKSendPaintEvents(const GdkRegion* region)
 #endif
 {
 #ifdef __WXGTK3__
+    if ( GetLayoutDirection() == wxLayout_RightToLeft )
     {
```

Comment:
I'm pretty sure setting the clip needs to stay.

Reply:
I tried these samples with the clip removed (LTR/RTL) :
- dataview
- drawing
- griddemo
- listctrl
- treectrl
- widgets

can you elaborate on why the clip should be altered at all so we can add a comment explaining why it should be done like that ?

Reply:
You can use `git blame` to identify the commit that added that code (bca7313499c11a6d7fecd2baa355ac09fd3ac83b in this case). Commit messages will often give you some context. I admit that code could probably use a comment. But in general, you should be very cautious about removing code.